simulator/func_sim/alu.h

```diff
@@ -308,6 +308,20 @@ struct ALU
     {
         instr->v_dst = sign_extension<bitwidth<T>>( instr->v_src1 + instr->v_imm);
     }
+
+    // Bit-field
+    template<typename I> static
+    void bit_field_place( I* instr)
+    {
+        using XLENType = typename I::RegisterUInt;
+        size_t XLEN = bitwidth<XLENType>;
+        int len = (instr->v_src2 >> 24) & 15;
```

Comment:
Please use `size_t` here. Also, you need a `narrow_cast` because v_src2 may be wider (128/64) than `int/size_t` (64/32).